### PR TITLE
Refactoring: Link polymorphic static typing

### PR DESCRIPTION
**Pull request by Treycos**
#### Description

Polymorphic TS refactoring of the `Link` component

Allows the component to be statically typed given the component passed

No behavior changes expected

Before:

image

After:

image
image
image

### CODE REVIEW

File: frontend/src/Components/Link/Link.tsx [hunk 1]
```diff
@@ -1,96 +1,87 @@
...
   SyntheticEvent,
   useCallback,
 } from 'react';
 import { Link as RouterLink } from 'react-router-dom';
 import styles from './Link.css';
 
-interface ReactRouterLinkProps {
-  to?: string;
-}
+export type LinkProps<C extends ElementType = 'button'> =
```
Comment (Treycos) [inline frontend/src/Components/Link/Link.tsx:11]: Couldn't do it with an `interface`

image

File: frontend/src/Components/Link/Link.tsx [hunk 2]
```diff
@@ -1,96 +1,87 @@
...
-    target,
-    type,
-    isDisabled,
-    noRouter = false,
-    onPress,
-    ...otherProps
-  } = props;
+export default function Link<C extends ElementType = 'button'>({
+  className,
+  component,
```
Comment (markus101) [inline frontend/src/Components/Link/Link.tsx:23]:
```suggestion
  component: Component = 'button',
```

File: frontend/src/Components/Link/Link.tsx [hunk 3]
```diff
@@ -1,96 +1,87 @@
...
+  to,
+  target,
+  type,
+  isDisabled,
+  noRouter,
+  onPress,
+  ...otherProps
+}: LinkProps<C>) {
+  const Cmp = component || 'button';
```
Comment (markus101) [inline frontend/src/Components/Link/Link.tsx:32]: Any reason we can't do this while destructuring? Also `Component` > `Cmp`

File: frontend/src/Components/Link/Link.tsx [hunk 4]
```diff
@@ -1,96 +1,87 @@
...
 
   const onClick = useCallback(
     (event: SyntheticEvent) => {
-      if (!isDisabled && onPress) {
-        onPress(event);
-      }
+      if (isDisabled) return;
```
Comment (markus101) [inline frontend/src/Components/Link/Link.tsx:36]:
```suggestion
      if (isDisabled) {
        return;
      }
```

**Review by markus101 — CHANGES_REQUESTED** [review]

File: frontend/src/Components/Link/Link.tsx (hunk 2 above)
Comment (Treycos) [reply frontend/src/Components/Link/Link.tsx:23]: Never really understood why but Typescript refuses this syntax

image

**Review by markus101 — APPROVED** [review]

**Pull request merged**